src/backend.c

```diff
@@ -440,28 +440,30 @@ static bool opencl_test_instruction (hashcat_ctx_t *hashcat_ctx, cl_context cont
 
 static bool read_kernel_binary (hashcat_ctx_t *hashcat_ctx, const char *kernel_file, size_t *kernel_lengths, char **kernel_sources, const bool force_recompile)
 {
-  FILE *fp = fopen (kernel_file, "rb");
+  HCFILE fp;
 
-  if (fp != NULL)
+  if (hc_fopen (&fp, kernel_file, "rb") != false)
```

Comment:
Please review all changes and try to -not- use negative comparison statements, in general try to avoid ! operator.

Replace with

```
if (hc_fopen (&fp, kernel_file, "rb") == true)
```